docs/user-guide/build-environment.md

```diff
@@ -0,0 +1,8 @@
+# Build Environment
+
+[Custom tools](config-management-plugins.md), [Helm](helm.md), and [Jsonnet](jsonnet.md) support the following build env vars:
+
+* `ARGOCD_APP_NAME` - name of application
+* `ARGOCD_APP_NAMESPACE` - destination application namespace.
+* `ARGOCD_APP_TARGET_REVISION` - the target revision from the spec, e.g. `master`.
+* `ARGOCD_APP_REVISION` - the resolved revision, e.g. `f913b6cbf58aa5ae5ca1f8a2b149477aebcbd9d8`
```

Comment:
Why only config managed plugins have access to system environment variables but not Helm and Jsonnet? I think we should be consistent and provide system environment variables to all tools.

Reply:
I don't believe they can consume them?

Reply:
Why not? In case of helm user might want to use 

```yaml
  spec:
    source:
      helm:
        parameters:
        - name: my-parameter
          value: $MY_ENV_VARIABLE
```

Reply:
ah - that is how they work - do you want a demo?